lib/iris/tests/unit/tests/stock/test_netcdf.py

```diff
@@ -0,0 +1,141 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""Unit tests for the :mod:`iris.tests.stock.netcdf` module."""
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import shutil
+import tempfile
+
+from iris import load_cube
+from iris.tests.stock import netcdf
+from iris.experimental.ugrid import MeshCoord, PARSE_UGRID_ON_LOAD
+
+
+class XIOSFileMixin(tests.IrisTest):
+    @classmethod
+    def setUpClass(cls):
+        # Create a temp directory for transient test files.
+        cls.temp_dir = tempfile.mkdtemp()
+
+    @classmethod
+    def tearDownClass(cls):
+        # Destroy the temp directory.
+        shutil.rmtree(cls.temp_dir)
+
+    def create_synthetic_file(self, **create_kwargs):
+        # Should be overridden to invoke one of the create_file_ functions.
+        # E.g.
+        # return create_file__xios_2d_face_half_levels(
+        #     temp_file_dir=self.temp_dir, dataset_name="mesh", **create_kwargs
+        # )
+        raise NotImplementedError
+
+    def create_synthetic_test_cube(self, **create_kwargs):
+        file_path = self.create_synthetic_file(**create_kwargs)
+        with PARSE_UGRID_ON_LOAD.context():
+            cube = load_cube(file_path)
+        return cube
+
+    def check_cube(self, cube, shape, location, level):
+        # Basic checks on the primary data cube.
+        self.assertEqual(cube.var_name, "thing")
+        self.assertEqual(cube.long_name, "thingness")
+        self.assertEqual(cube.shape, shape)
+
+        # Also a few checks on the attached mesh-related information.
+        last_dim = cube.ndim - 1
+        self.assertTrue(hasattr(cube, "mesh"))
```

Comment:
This bit isn't quite right.  A cube now _always_ has a "mesh" attribute.
You should probably check either that it is non-None, or maybe that it is a `Mesh` ?

Reply:
Ah! This has already caught me out somewhere else, and now I know why! Will fix...

Reply:
736a6216e095cb483314457d4ab8bedf3ef2fa7e